File: modules/day-01-readable/python/solution.py
student_names = ["Ana", "Ben", "Cam", "Dev", "Eli"]
exam_scores = [88, 91, 76, 95, 84]
attendance_flags = [1, 1, 0, 1, 1]
bonus_points = [2, 2, 0, 3, 1]


def final_score(score, attended_review, bonus):
    return (score * 0.8) + (attended_review * 10) + bonus


def performance_label(score):
    if score >= 90:
        return "did great"
    if score >= 80:
        return "ok"
    return "needs help"


student_results = []

for name, score, attended_review, bonus in zip(
    student_names, exam_scores, attendance_flags, bonus_points
):
    total = final_score(score, attended_review, bonus)
    student_results.append({"name": name, "final_score": total})

print("Student results")
for result in student_results:
    print(result["name"], performance_label(result["final_score"]))

average_score = sum(result["final_score"] for result in student_results) / len(student_results)
print("Average final score:", average_score)

File: modules/day-01-readable/r/solution.R
student_names <- c("Ana", "Ben", "Cam", "Dev", "Eli")
exam_scores <- c(88, 91, 76, 95, 84)
attendance_flags <- c(1, 1, 0, 1, 1)
bonus_points <- c(2, 2, 0, 3, 1)

gradebook <- data.frame(
  student_name = student_names,
  exam_score = exam_scores,
  attended_review = attendance_flags,
  bonus_points = bonus_points
)

gradebook$final_score <-
  (gradebook$exam_score * 0.8) +
  (gradebook$attended_review * 10) +
  gradebook$bonus_points

score_to_label <- function(score) {
  if (score >= 90) {
    return("did great")
  }

  if (score >= 80) {
    return("ok")
  }

  "needs help"
}

cat("Student results\n")

for (row_index in seq_len(nrow(gradebook))) {
  student <- gradebook$student_name[row_index]
  label <- score_to_label(gradebook$final_score[row_index])
  cat(student, label, "\n")
}

cat("Average final score:", mean(gradebook$final_score), "\n")
